### PR TITLE
Cleanup TikTorch menu in NN workflows

### PR DESCRIPTION
**Pull request by bitwizard25**
Related to #3156
- Remove non-functional menu items (Set hyperparameters, Create TikTorch
  configuration, Validation Set) along with their dead dialog classes
  (ParameterDlg, ValidationDlg)
- Rename menu from "TikTorch" to "Neural Network"
- Add "Open bioimageio cache folder" — opens the cache dir in the system
  file manager; informs user if it doesn't exist yet
- Add "Delete cache folder" — confirms before deleting via shutil.rmtree

## Checklist

- [ ] Format code and imports.
- [ ] Add docstrings and comments.
- [ ] Add tests.
- [ ] Reference relevant issues and other pull requests.
- [ ] Rebase commits into a logical sequence.
- [ ] Update [user documentation](https://github.com/ilastik/ilastik.github.io).
- [ ] Add screenshots / screen recordings.

### CODE REVIEW

**Comment by btbest** [comment]
Concerns:

- Relying on an explicitly internal module of bioimageio
- platformdirs fallback - is that actually bioimageio behaviour?
- tests with local import in each tests is a bit weird
- not sure about the testing approach patching the biomiageio dir when the main purpose of such a test would be to ensure the bioimageio dir hasn't diverged from where we're expecting it

**Comment by k-dominik** [comment]
sorry @bitwizard25, closing this as we needed it earlier now.

The part about adding UI/Option to delete the cache for bioimageio is still relevant - but pending updates to bioimageio.
